src/main/scala/org/ergoplatform/nodeView/mempool/ErgoMemPool.scala

```diff
@@ -105,6 +152,50 @@ class ErgoMemPool private[mempool](pool: OrderedTxPool)(implicit settings: ErgoS
       .map(_.value)
       .sum
 
+  /**
+    * Get average fee for the specified wait time interval
+    * @param expectedWaitTimeMinutes maximal amount of time for which transaction can be kept in mempool
+    * @param txSize size of transaction (in bytes)
+    *  @return recommended fee value for transaction to be proceeded in specified time
+    */
+  def getRecommendedFee(expectedWaitTimeMinutes : Int, txSize : Int) : Long = {
+    if (expectedWaitTimeMinutes < MemPoolStatisticsParams.nHistogramBins) {
+      // locate first non-empty histogram bin preceding or equal to the specified wait time
+      for (i <- expectedWaitTimeMinutes to 0 by -1) {
+        val h = stats.histogram(expectedWaitTimeMinutes)
+        if (h.nTxns != 0) {
+          return h.totalFee / h.nTxns * txSize / 1024
+        }
+      }
+    }
+    settings.nodeSettings.minimalFeeAmount
+  }
+
+  /**
+    * Calculate position in mempool corresponding to the specified fee and
+    * estimate time of serving this transaction based on average rate of placing
+    * transactions in blockchain
+    * @param txFee transaction fee
+    * @param txSize size of transaction (in bytes)
+    * @return average time for this transaction to be placed in block
+    */
+  def getExpectedWaitTime(txFee : Long, txSize : Int): Long  = {
+    // Create dummy transaction entry
+    val feePerKb = txFee * 1024 / txSize
+    val dummyModifierId = bytesToId(Array.fill(32)(0.toByte))
+    val wtx = WeightedTxId(dummyModifierId, feePerKb, feePerKb, 0)
+
+    // Find position of entry in mempool
+    val posInPool = pool.orderedTransactions.keySet.until(wtx).size
+
+    // Time since statistics measurement interval (needed to calculate average tx rate)
+    val elapsed = System.currentTimeMillis() - stats.startMeasurement
+    if (stats.takenTxns != 0) {
+      elapsed * posInPool / stats.takenTxns
+    } else {
+      0
+    }
+  }
```

Comment:
Can be moved out to a histogram builder and then unit-tested independently of this mempool, in particular using mock time provider.